pkg/operatorstatus/operatorstatus.go

```diff
@@ -59,40 +59,47 @@ type ClusterOperatorWrapper interface {
 }
 
 func (o *ClusterOperatorHandler) UpdateOperatorStatus(sampleResource *v1alpha1.SamplesResource) error {
+
 	var err error
+
 	switch {
 	case sampleResource.ConditionTrue(v1alpha1.SamplesExist):
-		err = o.setOperatorStatus(configv1.OperatorAvailable, configv1.ConditionTrue, "Samples exist in the openshift project")
+		err = o.setOperatorStatus(configv1.OperatorAvailable, configv1.ConditionTrue, fmt.Sprintf("Samples installed at %s", v1alpha1.CodeLevel))
 	case sampleResource.ConditionFalse(v1alpha1.SamplesExist):
-		err = o.setOperatorStatus(configv1.OperatorAvailable, configv1.ConditionFalse, "Samples do not exist in the openshift project")
+	case sampleResource.ConditionUnknown(v1alpha1.SamplesExist):
+		err = o.setOperatorStatus(configv1.OperatorAvailable, configv1.ConditionFalse, fmt.Sprintf("Samples not installed at %s", v1alpha1.CodeLevel))
 	default:
-		err = o.setOperatorStatus(configv1.OperatorAvailable, configv1.ConditionUnknown, "The presence of the samples in the openshift project is unknown")
+		err = o.setOperatorStatus(configv1.OperatorAvailable, configv1.ConditionUnknown, fmt.Sprintf("Samples at %s unknown", v1alpha1.CodeLevel))
 	}
 	if err != nil {
 		return err
 	}
 
+	errorDetails := sampleResource.ConditionsMessages()
 	switch {
-	case sampleResource.ConditionTrue(v1alpha1.ConfigurationValid):
-		err = o.setOperatorStatus(configv1.OperatorFailing, configv1.ConditionFalse, "The samples operator configuration is valid")
+	// errorDetails should have data if config valid false or progress unknown, but just in case
 	case sampleResource.ConditionFalse(v1alpha1.ConfigurationValid):
-		err = o.setOperatorStatus(configv1.OperatorFailing, configv1.ConditionTrue, "The samples operator configuration is invalid")
+	case sampleResource.ConditionUnknown(v1alpha1.ImageChangesInProgress):
+	case len(errorDetails) > 0:
+		err = o.setOperatorStatus(configv1.OperatorFailing, configv1.ConditionTrue, fmt.Sprintf("Unable to install samples at %s: %s", v1alpha1.CodeLevel, errorDetails))
+		if err != nil {
+			return err
+		}
+		// on errors we stop after this setting
+		return o.setOperatorStatus(configv1.OperatorProgressing, configv1.ConditionTrue, fmt.Sprintf("Samples unable to reach %s", v1alpha1.CodeLevel))
```

Comment:
is it "unable to reach" or is it "Samples are updating to....."?  if it's "unable to reach" i'd expect us to be in failing=true and progressing=false, but this is setting progressing=true.

Reply:
(did you mean to be setting the Failing condition here instead of Progressing?)